drivers/volume/portworx/portworx.go

```diff
@@ -648,6 +650,78 @@ func (d *portworx) WaitForUpgrade(n node.Node, image, tag string) error {
 	return nil
 }
 
+func (d *portworx) GetReplicationFactor(vol *torpedovolume.Volume) (int64, error) {
+	name := d.schedOps.GetVolumeName(vol)
+	t := func() (interface{}, bool, error) {
+		vols, err := d.volDriver.Inspect([]string{name})
+		if err != nil && err == volume.ErrEnoEnt {
+			return 0, false, volume.ErrEnoEnt
+		} else if err != nil {
+			return 0, true, err
+		}
+		if len(vols) == 1 {
+			return vols[0].Spec.HaLevel, false, nil
+		}
+		return 0, false, fmt.Errorf("Extra volumes with the same volume name/ID seen") //Shouldn't reach this line
+	}
+
+	iReplFactor, err := task.DoRetryWithTimeout(t, validateReplicationUpdateTimeout, defaultRetryInterval)
+	if err != nil {
+		return 0, &ErrFailedToGetReplicationFactor{
+			ID:    name,
+			Cause: err.Error(),
+		}
+	}
+	replFactor, ok := iReplFactor.(int64)
+	if !ok {
+		return 0, &ErrFailedToGetReplicationFactor{
+			ID:    name,
+			Cause: fmt.Sprintf("Replication factor is not of type int64"),
+		}
+	}
+
+	return replFactor, nil
+}
+
+func (d *portworx) SetReplicationFactor(vol *torpedovolume.Volume, replFactor int64) error {
+	name := d.schedOps.GetVolumeName(vol)
+	t := func() (interface{}, bool, error) {
+		vols, err := d.volDriver.Inspect([]string{name})
+		if err != nil && err == volume.ErrEnoEnt {
+			return nil, false, volume.ErrEnoEnt
+		} else if err != nil {
+			return nil, true, err
+		}
+
+		if len(vols) == 1 {
+			spec := &api.VolumeSpec{
+				HaLevel:          int64(replFactor),
+				SnapshotInterval: math.MaxUint32,
+				ReplicaSet:       &api.ReplicaSet{},
+			}
+			locator := &api.VolumeLocator{
+				Name:         vols[0].Locator.Name,
+				VolumeLabels: vols[0].Locator.VolumeLabels,
+			}
+			err = d.volDriver.Set(vols[0].Id, locator, spec)
+			if err != nil {
+				return nil, false, err
```

Comment:
should retry on this failure

Reply:
Why should this be retried? This is the failure of setting repl factor. Should flag it is an error and be triaged. If anything, overtime we may come up with a list of retry-able exceptions. Not sure we are there yet.

Reply:
No need to retry now. If we hit intermittent errors in future in `Set(...)` we will evaluate whether to retry or not at that point.